pkg/controller/migmigration/task.go

```diff
@@ -1397,7 +1432,7 @@ func (t *Task) anyFlags(phase Phase) (bool, error) {
 	if phase.any&IndirectImage != 0 && t.indirectImageMigration() {
 		return true, nil
 	}
-	if phase.any&EnableImage != 0 && !t.PlanResources.MigPlan.IsImageMigrationDisabled() {
+	if phase.any&EnableImage != 0 && !t.PlanResources.MigPlan.IsImageMigrationDisabled() && t.migrateState() {
```

Comment:
I don't think you want this. I don't think `t.migrateState()` should be considered for the `EnableImage` test.

Reply:
@sseago since we are using Stage itinerary for State Migrations & Storage Conversion, we want this. We want to disable image migration whenever any of the State Migrations or Storage Conversion is set.